src/server/pages/home-page.jsx

```diff
@@ -12,7 +16,93 @@ import TrendingNotebooksList from "../components/trending-notebooks-list";
 import AttentionBlock from "../components/attention-block";
 import NewNotebookButton from "../components/new-notebook-button";
 import FeaturedNotebooks from "../../shared/components/featured-notebooks";
+import DropTarget from "../../shared/components/drop-target";
+import THEME from "../../shared/theme";
 import { sharedProperties } from "../../server/style/base";
+import { createNotebookRequest } from "../../shared/server-api/notebook";
+import { saveFileToServer } from "../../shared/utils/file-operations";
+import { fadeIn, bounce } from "../../shared/keyframes";
+
+const Overlay = styled.div`
+  align-items: center;
+  animation: ${fadeIn} 0.25s;
+  background: ${THEME.header.backgroundLeft};
+  box-sizing: border-box;
+  color: #fff;
+  display: grid;
+  grid-gap: 100px;
+  justify-items: center;
+  overflow: auto;
+  padding: 2%;
+  z-index: 10;
+
+  /* Expand to fill the viewport */
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+
+  &.invisible {
+    display: none;
+  }
+`;
+
+// This wrapper works around a peculiarity with centering in CSS Grid.
+//
+// Setting "align-content: center" on a grid container gives it "true" vertical
+// centering, which is to say that content will be clipped it it cannot fit on
+// screen, even when "overflow: auto" is used.
+//
+// In other words, we could ditch this wrapper and set "align-content: center"
```

Comment:
Hmm... I'll have to think about this. The file modal doesn't impose any limit. I don't think the workaround is too crazy: just one extra DOM element. Are we worried about the load on the server?

Reply:
Sure. I'll add a warning if more than 8 files are dragged and open an issue to set some limit in the back-end. I think it would still be good to keep this wrapper, though. It's not so much about the number of files as it is about how many filenames can be shown on screen at one time. Even if we limit to 8, we'd want this wrapper in the case that someone has a large font or a short window.